src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,55 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  isAlive() {
+    return this.health > 0;
+  }
+
+  die() {
+    this.health = 0;
+
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+
+    return `${this.name} is now hidden from carnivores.`;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
```

Comment:
Ensure that the `herbivore` parameter is indeed an instance of `Herbivore` and not null or undefined before accessing its properties. This will prevent potential runtime errors.